src/App.tsx

```diff
@@ -26,70 +27,126 @@ type ReorderOptions = {
+  visibleGoods.sort((a, b) => {
+    switch (sortType) {
+      case SortType.ALPABET:
+        return a.localeCompare(b);
+
+      case SortType.LENGTH:
+        return a.length - b.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    sortType: SortType.NONE,
+    isReversed: false,
+  };
+
+  sortByAlph = () => {
+    this.setState({
+      sortType: SortType.ALPABET,
+    });
+  };
+
+  sortByLength = () => {
+    this.setState({
+      sortType: SortType.LENGTH,
+    });
+  };
+
+  goodsReversed = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  reset = () => {
+    this.setState({
+      sortType: SortType.NONE,
+      isReversed: false,
+    });
+  };
+
+  render() {
+    const { sortType, isReversed } = this.state;
+    const goods = getReorderedGoods(goodsFromServer, { sortType, isReversed });
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={className('button is-info', {
+              'is-light': sortType !== SortType.ALPABET,
+            })}
+            onClick={this.sortByAlph}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={className('button is-success', {
+              'is-light': sortType !== SortType.LENGTH,
+            })}
+            onClick={this.sortByLength}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={className('button is-warning', {
+              'is-light': !isReversed,
+            })}
+            onClick={this.goodsReversed}
+          >
+            Reverse
+          </button>
+
+          <button
+            type="button"
+            className={className('button is-danger is-light', {
```

Comment:
Just don't render this button if nothing happened
Use [conditional rendering](https://reactjs.org/docs/conditional-rendering.html)
That's why tests fail
![image](https://user-images.githubusercontent.com/96409563/203936829-4b890f0a-9cf9-4cb1-88df-5c32e65316c6.png)